### PR TITLE
Upload sprint results to fragalysis

### PR DESCRIPTION
**Pull request by glass-w**
## Description
Provides uploads to Fragalysis

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] Setup new function to push to Fragalysis.
  - [x] Ensure formatting is correct for Fragalysis files.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by glass-w** [comment]
This now uses a `fragalysis_config.json` file to specify paramters needed by the fragalysis API (see [here](https://discuss.postera.ai/t/providing-computed-poses-for-others-to-look-at/1155/8) and [here](https://github.com/xchem/fragalysis-api/blob/master/fragalysis_api/xcextracter/computed_set_update.py#L34)). 

An example of this type of file:

```json
{
        "run": true,
        "ligands_filename": "reliable-transformations-final-ligands.sdf",
        "proteins_filename": "reliable-transformations-final-proteins.pdb",
        "fragalysis_sdf_filename": "compound-set_foldingathome-sprint-5.sdf",
        "ref_url": "https://discuss.postera.ai/t/folding-home-sprint-5/2423",
        "ref_mols": "x12073",
        "ref_pdb": "x12073",
        "target_name": "MPro",
        "submitter_name": "Folding@home",
        "submitter_email": "first.last@choderlab.org",
        "submitter_institution": "MSKCC",
        "method": "Sprint 5",
        "upload_key": "<key number>",
        "new_upload": true 
}
```

**Comment by glass-w** [comment]
Currently this doesn't appear to upload results to the Fragalysis site but runs without errors. For now this can safely be put into production as long as either a) the `fragalysis-config` flag isn't specified or b) `run` is set to `true` in the `fragalysis_config.json` file (as described above).

File: environment.yml [hunk 1]
```diff
@@ -19,11 +20,16 @@ dependencies:
   - pydantic
   - pymbar
   - python >= 3.6
+  - pymol
```
Comment (jchodera) [inline environment.yml:23]: Wow, do we need pymol?!

File: fah_xchem/analysis/report.py [hunk 2]
```diff
@@ -132,6 +137,142 @@ def RenderData(image, mol, tags):
...
+    molecules = []
+
+    with oechem.oemolistream(ligands_path) as ifs:
+        oemol = oechem.OEGraphMol()
+        while oechem.OEReadMolecule(ifs, oemol):
+            molecules.append(oemol.CreateCopy())
+    print(f"{len(molecules)} ligands read")
+
+    descriptions = {
+        "DDG (kcal/mol)": "Relative computed free energy difference",
```
Comment (jchodera) [inline fah_xchem/analysis/report.py:190]: We might eventually update this to DG and put the absolute free energy and predicted pIC50, but this is great for now.

File: environment.yml (hunk 1 above)
Comment (jchodera) [reply environment.yml:23]: It looks like we don't use it---should we remove this and the `schrodinger` channel?

File: environment.yml [hunk 3]
```diff
@@ -19,11 +20,16 @@ dependencies:
...
+  - rdkit
```
Comment (jchodera) [inline environment.yml:24]: Do we explicitly use rdkit?

**Review by jchodera — APPROVED** [review]
Looks to be missing some example input---can you add this somewhere, or to the `README.md`? Also, it looks like we added some dependencies we don't use (pymol and rdkit).

File: environment.yml (hunk 3 above)
Comment (glass-w) [reply environment.yml:24]: It looks like rdkit is used for `xcimporter.py` in the fragalysis api. I'll check if it is needed for this pipeline to work and remove it (and pymol) if not.

File: environment.yml (hunk 1 above)
Comment (glass-w) [reply environment.yml:23]: I'll double check, if we don't need it I'll remove it before merging

File: environment.yml (hunk 3 above)
Comment (glass-w) [reply environment.yml]: Removed

File: environment.yml (hunk 1 above)
Comment (glass-w) [reply environment.yml]: Removed

**Review by jchodera — APPROVED** [review]
Looks good! Shold we merge now and disable uploads until we can figure out what is going wrong.

**Comment by glass-w** [comment]
@jchodera I'm going to see if I can fix the zipped PDB files first. If this takes too long we can merge anyway and just disable the upload.

**Comment by glass-w** [comment]
Zipped PDB files now fixed and upload to Fragalysis is working.

**Review by jchodera — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**